bciers/apps/reporting/src/app/components/changeReview/templates/facilityReportOrganizer.ts

```diff
@@ -0,0 +1,354 @@
+import { ChangeItem, FacilityReportStructure } from "../constants/types";
+import { parseFacilityReportField } from "./facilityReportParser";
+
+function handleFuelsData(
+  unit: any,
+  fuelsData: any[],
+  change: ChangeItem,
+  sourceTypeName: string,
+  unitIndex: number,
+) {
+  fuelsData.forEach((fuelData: any, fuelIndex: number) => {
+    if (!unit.fuels[fuelIndex]) {
+      unit.fuels[fuelIndex] = {
+        fuelIndex,
+        fields: [],
+        emissions: {},
+      };
+    }
+
+    const fuel = unit.fuels[fuelIndex];
+
+    // Add fuel-level fields
+    Object.entries(fuelData).forEach(([key, value]) => {
+      if (key !== "emissions") {
+        fuel.fields.push({
+          field: `${change.field}[source_types][${sourceTypeName}][units][${unitIndex}][fuels][${fuelIndex}][${key}]`,
+          old_value: null,
+          new_value: value,
+          change_type: "added",
+        });
+      }
+    });
+
+    // Handle fuel emissions
+    if (fuelData.emissions && Array.isArray(fuelData.emissions)) {
+      fuelData.emissions.forEach((emissionData: any, emissionIndex: number) => {
+        if (!fuel.emissions[emissionIndex]) {
+          fuel.emissions[emissionIndex] = {
+            emissionIndex,
+            fields: [],
+          };
+        }
+
+        const emission = fuel.emissions[emissionIndex];
+        Object.entries(emissionData).forEach(([key, value]) => {
+          emission.fields.push({
+            field: `${change.field}[source_types][${sourceTypeName}][units][${unitIndex}][fuels][${fuelIndex}][emissions][${emissionIndex}][${key}]`,
+            old_value: null,
+            new_value: value,
+            change_type: "added",
+          });
+        });
+      });
+    }
+  });
+}
+
+function handleUnitEmissionsData(
+  unit: any,
+  emissionsData: any[],
+  change: ChangeItem,
+  sourceTypeName: string,
+  unitIndex: number,
+) {
+  emissionsData.forEach((emissionData: any, emissionIndex: number) => {
+    if (!unit.emissions[emissionIndex]) {
+      unit.emissions[emissionIndex] = {
+        emissionIndex,
+        fields: [],
+      };
+    }
+
+    const emission = unit.emissions[emissionIndex];
+    Object.entries(emissionData).forEach(([key, value]) => {
+      emission.fields.push({
+        field: `${change.field}[source_types][${sourceTypeName}][units][${unitIndex}][emissions][${emissionIndex}][${key}]`,
+        old_value: null,
+        new_value: value,
+        change_type: "added",
```

Comment:
I think I'm missing something - how do we know that the emission has been added?